src/internal_modules/roc_core/pool_impl.h

```diff
@@ -30,10 +30,39 @@ namespace core {
 //!
 //! @see Pool.
 class PoolImpl : public NonCopyable<> {
+private:
+    struct Slab : ListNode {};
+    struct Slot : ListNode {};
+
+    // Slot layout when supplying data to user.
+    struct UserSlot {
+        PoolImpl* owner;
+        AlignMax canary_before;
+        AlignMax data[];
+        // Another canary immediately after actual data. Any padding for data is
+        // considered part of canary.
+    };
```

Comment:
nitpick: private definitions should go after public.

I'm always forgetting to add it to code style doc, but the order is usually:

* public
  * types & constants
  * methods
* private
  * types & constants
  * methods
  * fields 

This is not super-strict, but preferred for consistency.

Reply:
Will fix. It was because I needed the definitions before the complicated SlotSize.